src/coreclr/gc/gc.cpp

```diff
@@ -40344,7 +40344,12 @@ HRESULT GCHeap::Initialize()
 #ifdef MULTIPLE_HEAPS
     gc_heap::soh_segment_size /= 4;
 #endif //MULTIPLE_HEAPS
-    gc_heap::min_segment_size_shr = index_of_highest_set_bit (REGION_SIZE);
+    size_t gc_region_size = (size_t)GCConfig::GetGCRegionsSize();
+    if (!power_of_two_p(gc_region_size) || gc_region_size * nhp * 10 > gc_heap::regions_range)
```

Comment:
I'm wondering why this is * 10 :) we would need to have at least 19 (basic) regions per heap if we want to make sure we can initialize successfully -

3 (3 regions for gen0/1/2 each) + 8 (one region for LOH) + 8 (one region for POH) = 19

Reply:
also nit 

` (!power_of_two_p(gc_region_size) || gc_region_size * nhp * 10 > gc_heap::regions_range)`
->
` (!power_of_two_p(gc_region_size) || ((gc_region_size * nhp * 10) > gc_heap::regions_range))`

I'll continue to point these nit things out till they stick :)